tools/windows-node-installer/test/e2e/aws_test.go

```diff
@@ -1,90 +1,289 @@
+// Instance input needs to be updated before use.
+func TestInstanceInfraTagExists(t *testing.T) {
 	key := "kubernetes.io/cluster/" + infraID
 	value := "owned"
-	for _, tag := range instance.Tags {
+	for _, tag := range createdInstance.Tags {
 		if *tag.Key == key && *tag.Value == value {
-			return true
+			return
+		}
+	}
+	assert.Fail(t, "infrastructure tag not found")
+}
+
+// TestInstanceClusterWorkerSg asserts if the created instance has OpenShift cluster worker's security group attached.
+func TestInstanceClusterWorkerSg(t *testing.T) {
+	workerSg, err := awsProvider.GetClusterWorkerSGID(infraID)
+	assert.NoError(t, err, "failed to get OpenShift cluster worker security group")
+
+	for _, sg := range createdInstance.SecurityGroups {
+		if *sg.GroupId == workerSg {
+			return
 		}
 	}
-	return false
+	assert.Fail(t, "instance is not associated with OpenShift cluster worker security group")
+}
+
+// TestInstanceIsAssociatedWithWindowsWorkerSg asserts if the created instance has a security group made for the Windows
+// instance attached by checking the group name, recorded id, necessary ports, and ip-permission.cidr.
+func TestInstanceIsAssociatedWithWindowsWorkerSg(t *testing.T) {
+	myIp, err := awscp.GetMyIp()
+	assert.NoError(t, err, "error getting user's public IP")
+
+	vpc, err := awsProvider.GetVPCByInfrastructure(infraID)
+	assert.NoError(t, err, "error getting OpenShift cluster VPC")
+
+	sgs, err := awsProvider.EC2.DescribeSecurityGroups(&ec2.DescribeSecurityGroupsInput{
+		Filters: []*ec2.Filter{
+			{
+				Name:   aws.String("ip-permission.from-port"),
+				Values: aws.StringSlice([]string{"-1", "3389"}),
+			},
+			{
+				Name:   aws.String("ip-permission.to-port"),
+				Values: aws.StringSlice([]string{"-1", "3389"}),
+			},
+			{
+				Name:   aws.String("ip-permission.protocol"),
+				Values: aws.StringSlice([]string{"tcp"}),
+			},
+			{
+				Name:   aws.String("group-id"),
+				Values: aws.StringSlice([]string{createdSgID}),
+			},
+			{
+				Name:   aws.String("ip-permission.cidr"),
+				Values: aws.StringSlice([]string{myIp + "/32", *vpc.CidrBlock}),
+			},
+		},
+	})
+	if err != nil || len(sgs.SecurityGroups) < 1 {
+		assert.Fail(t, "instance is not associated with a Windows security group, %v", err)
+	}
+}
+
+// TestInstanceIsAssociatedWithClusterWorkerIAM asserts if the created instance has the OpenShift cluster worker's IAM
+// attached.
+func TestInstanceIsAssociatedWithClusterWorkerIAM(t *testing.T) {
+	iamProfile, err := awsProvider.GetIAMWorkerRole(infraID)
+	assert.NoError(t, err, "error getting OpenShift Cluster Worker IAM")
+
+	assert.Equal(t, *iamProfile.Arn, *createdInstance.IamInstanceProfile.Arn, "instance is not associated with worker IAM profile")
+}
+
+// TestDestroyWindowsInstance terminates the created Windows instance.
+func TestDestroyWindowsInstance(t *testing.T) {
+	err := awsProvider.DestroyWindowsVMs()
+	assert.NoError(t, err, "error destroying instance")
```

Comment:
You need to check if instance has been deleted here and not in another test case. Same goes for SG changes.